src/templates/base/2019/contributors.html

```diff
@@ -362,12 +385,19 @@ <h1 class="title title-lg">
     var totalText = document.querySelector('#contributors-total-text');
     var totalContributors = document.querySelector('#contributors-total');
     var shadowNode = document.createElement('div');
-    var urlObject = new URL(document.location);
+    var urlObject;
+    
+    if (self.URLSearchParams) {
```

Comment:
> Are we explicitly trying to support IE? Why?

It's not so much that I'm explicitly trying to support IE - I just see it as a proxy for "old browser" and still do test code in that, so if we can without too much effort than that's a code thing in my mind. It's defensive coding so other old browsers don't bomb out. So I try to avoid `if IE11` as much as possible (though have put a few of those in, in the past admittedly), however I do like `if (self.XXX)` code for any relatively new feature.

I appreciate the vast majority of our audience is likely to be on modern browsers, but do try best to support all browsers as much as possible (as long as it doesn't make our code very messy to do that) in the aims of making the site as accessible to all as possible. I noticed the current production was broken when testing in IE11 and decided to include the necessary checks in this PR.

Maybe I'm being OTT (again!) ¯\_(ツ)_/¯

> Why self as opposed to window?

To be honest I thought that was [good practice](https://developer.mozilla.org/en-US/docs/Web/API/Window/self). Is it not and more confusing? If so happy to switch.

> Why URLSearchParams as opposed to URL?

Well that's an interesting one. IE (plus a few other older browsers) supports `URL` but not the [URL constructor](https://developer.mozilla.org/en-US/docs/Web/API/URL#Browser_compatibility). So just using `self.URL` still threw the exception in IE. Wasn't sure how to test for just the constructor's support (`self.URL()` doesn't work). And we need `URLSearchParams` later anyway and every browser that supports `URLSearchParams` supports `URL()`. 

However now [found out how to do this properly](https://stackoverflow.com/questions/44363132/how-to-detect-if-url-constructor-is-supported-by-browsers) so will change to that:

```javascript
    if (typeof URL === "function") {
      urlObject = new URL(document.location);
    }
```

Reply:
Fixed.

Reply:
For another issue, what are your thoughts on transpilation? It'd be great if developers could write using the latest JS features and not have to worry about browser support.

Reply:
Not used them much to be honest. But then I don't get a chance to code JavaScript much except in this project. Personally I do like having the code in production as close as source. But that's cause I come from a production support background so have spent far too much of my time debugging in prod, rather than in dev. And too few developers use source maps (or they don't work).

Personally I'm leaning to the "don't really need the complexity". We're a static website basically, and not a web app. And not that complex one to be honest. Then again, most web apps don't try very hard to support old browsers either as quickly becomes a losing game, whereas it is much more possible for static web sites so maybe I'm looking at it wrong and websites are were transpilation adds the real benefit because they can degrade much more gracefully 😀

Reply:
Ok thanks for weighing in. Would you be open to creating a new issue to explore the idea with the rest of the dev team?

Reply:
Will do

Reply:
Btw, the reason I being so defensive in the coding here is because if there are any exceptions at all, then the JavaScript stops and the final `data-rendered` attribute is not applied, and so no contributors show at all. That’s not good. So that’s what I’m trying to avoid. IE is still not supported - the filter buttons don’t work, but at least it doesn’t show a blank page anymore.

Maybe, instead of all the checks, we should just have a try/catch round it so we can still show the contributors if something fails?

Reply:
> Maybe, instead of all the checks, we should just have a try/catch round it so we can still show the contributors if something fails?

💯 that sounds much better

Reply:
OK that's done. I still had to remove some of the rest parameters as that's syntax is not recognised by non-ES2015 browsers but over all it's a much smaller, and cleaner, change now. Thanks for working through this with me! 👍